nbextensions/bulkImportCell/bulkImportCell.js

```diff
@@ -608,37 +612,69 @@ define([
             const meta = cell.metadata;
             meta.kbase.attributes.lastLoaded = new Date().toUTCString();
             cell.metadata = meta;
-            render().then(() => {
-                jobManager.addHandler('modelUpdate', {
-                    execMessage: (jobManagerContext) => {
-                        // Update the execMessage panel with details of the active jobs
-                        controlPanel.setExecMessage(
-                            Jobs.createCombinedJobState(
+            render()
+                .then(() => {
+                    jobManager.addHandler('modelUpdate', {
+                        execMessage: (jobManagerContext) => {
+                            // Update the execMessage panel with details of the active jobs
+                            controlPanel.setExecMessage(
+                                Jobs.createCombinedJobState(
+                                    jobManagerContext.model.getItem('exec.jobs.byStatus')
+                                )
+                            );
+                        },
+                        fsmState: (jobManagerContext) => {
+                            const fsmState = Jobs.getFsmStateFromJobs(
                                 jobManagerContext.model.getItem('exec.jobs.byStatus')
-                            )
-                        );
-                    },
-                    fsmState: (jobManagerContext) => {
-                        const fsmState = Jobs.getFsmStateFromJobs(
-                            jobManagerContext.model.getItem('exec.jobs.byStatus')
-                        );
-                        if (fsmState) {
-                            updateState(fsmState);
+                            );
+                            if (fsmState) {
+                                updateState(fsmState);
+                            }
+                        },
+                    });
+                    try {
+                        jobManager.restoreFromSaved();
+                    } catch (e) {
+                        console.error(e);
+                    }
+                    const expectedFiles = new Set();
+                    Object.values(model.getItem('inputs')).forEach((inputs) => {
+                        for (const f of inputs.files) {
+                            expectedFiles.add(f);
                         }
-                    },
-                });
-
-                // TODO: assess cell state, update job info if required
-                try {
-                    jobManager.restoreFromSaved();
-                } catch (e) {
-                    console.log(e);
-                }
+                    });
+                    return BulkImportUtil.getMissingFiles(Array.from(expectedFiles)).catch(
```

Comment:
Added the `.catch` to this chunk of the Promise (rather than the next part of the Promise chain) so that it'll more easily pass on a silent empty Array if it fails.

I figure that if the staging service is down, bulk import cells should still be able to load their last known good state.